chia-wallet/src/derive_synthetic.rs

```diff
@@ -0,0 +1,125 @@
+use chia_bls::{PublicKey, SecretKey};
+use num_bigint::BigInt;
+use sha2::{digest::FixedOutput, Digest, Sha256};
+
+pub trait DeriveSynthetic {
+    fn derive_synthetic(&self, hidden_puzzle_hash: &[u8; 32]) -> Self;
+}
+
+impl DeriveSynthetic for PublicKey {
+    fn derive_synthetic(&self, hidden_puzzle_hash: &[u8; 32]) -> Self {
+        self + &synthetic_offset(self, hidden_puzzle_hash).public_key()
+    }
+}
+
+impl DeriveSynthetic for SecretKey {
+    fn derive_synthetic(&self, hidden_puzzle_hash: &[u8; 32]) -> Self {
+        self + &synthetic_offset(&self.public_key(), hidden_puzzle_hash)
+    }
+}
+
+pub fn hash_to_secret_key(bytes: [u8; 32]) -> SecretKey {
+    let value = BigInt::from_signed_bytes_be(bytes.as_slice());
+    let group_order = BigInt::parse_bytes(
+        b"73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001",
+        16,
+    )
+    .unwrap();
+    let modulo = ((&value % &group_order) + &group_order) % &group_order;
```

Comment:
I would expect you could move `value` into this expression, but perhaps the operator doesn't implement that variant